react-file-uploads/server/index.js

```diff
@@ -0,0 +1,74 @@
+require('dotenv/config');
+const path = require('path');
+const pg = require('pg');
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+const uploadsMiddleware = require('./uploads-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+const publicPath = path.join(__dirname, 'public');
+
+if (process.env.NODE_ENV === 'development') {
+  app.use(require('./dev-middleware')(publicPath));
+}
+
+app.use(express.static(publicPath));
+app.use(express.json());
+
+app.post('/api/uploads', uploadsMiddleware, (req, res, next) => {
+  console.log(res.body);
+  const { caption } = req.body;
+  if (!caption) {
+    throw new ClientError(400, 'caption is a required field');
+  }
+  const imageUrl = path.join('/images', req.file.filename);
+  const sql = `
+      insert into "images" ("caption", "url")
+              values ($1, $2)
+              returning *
+      `;
+  const params = [caption, imageUrl];
+  db.query(sql, params)
+    .then(result => {
+      const [image] = result.rows;
+      if (!image) {
+        throw new ClientError(401, 'invalid image upload');
+      }
```

Comment:
If the insert succeeded at all, then `result.rows[0]` is guaranteed to exist since you have a `returning` clause. Try to avoid including logic that you can't test.